packages/typescript-estree/tools/test-utils.ts

```diff
@@ -84,3 +68,67 @@ export function isJSXFileType(fileType: string): boolean {
   }
   return fileType === 'js' || fileType === 'jsx' || fileType === 'tsx';
 }
+
+/**
+ * Removes the given keys from the given AST object recursively
+ * @param root A JavaScript object to remove keys from
+ * @param keysToOmit Names and predicate functions use to determine what keys to omit from the final object
+ * @param nodes advance ast modifications
+ * @returns formatted object
+ */
+export function omitDeep<T = Record<string, unknown>>(
+  root: T,
+  keysToOmit: { key: string; predicate: (value: unknown) => boolean }[] = [],
+  nodes: Record<string, (node: T, parent: T | null) => void> = {},
+): T {
```

Comment:
This shouldn't be tied to the parent generic, because it makes it look like there are no mutations applied.
Doing so is very, very unsafe for obvious reasons.

```suggestion
): Record<string, unknown> {
```